Makefile

```diff
@@ -1,13 +1,19 @@
 # this is licensed software, @see LICENSE file.
+
+# OpenCV & Tensorflow recommended flags for performance..
 CFLAGS = -Ofast -march=native -fno-trapping-math -fassociative-math -funsafe-math-optimizations -Wall -pthread
 LDFLAGS = -lrt -ldl
 
+# Version
+VERSION=$(shell git describe --all --always --dirty)
```

Comment:
Why not `git describe --tags --dirty`? Gives much more descriptive version strings IMHO.

Reply:
I tried that, but without tags (or with tags in much older shared history) it's less useful (no branch indication, possibly confusing old tag base). This will display a tag if available in the current branch, or the branch itself. We could add `--long` to always show the commit id and revision count from the branch/tag start?

Reply:
Moved to #68,